test: do not run test-dlopen-so if tpm is disabled

eworm-de opened the pull request:
The test tries to dlopen `libtss2-esys.so.0`, and fails if that is not available. Make the test depend on ENABLE_TPM.

DaanDeMeyer commented:
This doesn't seem right, we should do feature checks inside the test, not in the build system, this fix will skip all the other dlopen tests as well if ENABLE_TPM is false. Can we get some more information? I guess this is trying to build and run the tests on Arch? Is tpm2-tss installed? If it isn't, the tpm2 dlopen test shouldn't be running, if it is, the so should be available and the test should succeed.

eworm-de commented:
Ah, right... I should have looked into the test itself. 😝

Some details about the background: The issue happens when building the Arch multilib package, where no tpm packages are available.

Changed the test itself to just skip that library if ENABLE_TPM is not set.

eworm-de commented:
This is the output of configuration:
```
==> Starting build()...
+ exec meson setup --prefix /usr --libexecdir lib --sbindir bin --buildtype plain --auto-features enabled --wrap-mode nodownload -D b_lto=true -D b_pie=true systemd-stable build --libexecdir /usr/lib32 --libdir /usr/lib32 -Dversion-tag=252.1-1-arch -Dshared-lib-tag=252.1-1 -Dmode=release -Daudit=false -Dblkid=false -Dgnu-efi=false -Dima=false -Dkmod=false -Dlibcryptsetup=false -Dlibidn2=true -Dlibiptc=false -Dlz4=false -Dmicrohttpd=false -Dpam=false -Dseccomp=false -Dutmp=false -Dhibernate=false -Dldconfig=false -Dresolve=true -Defi=false -Dtpm=false -Denvironment-d=false -Dbinfmt=false -Drepart=false -Dcoredump=false -Dpstore=false -Doomd=false -Dlogind=false -Dhostnamed=false -Dlocaled=false -Dmachined=true -Dportabled=false -Dsysext=false -Duserdb=false -Dhomed=false -Dnetworkd=false -Dtimedated=false -Dtimesyncd=false -Dremote=false -Dcreate-log-dirs=false -Dnss-myhostname=true -Dnss-mymachines=true -Dnss-resolve=true -Dnss-systemd=true -Dfirstboot=false -Drandomseed=false -Dbacklight=false -Dvconsole=false -Dquotacheck=false -Dsysusers=false -Dtmpfiles=false -Dimportd=false -Dhwdb=false -Drfkill=false -Dxdg-autostart=false -Dman=false -Dhtml=false -Dtranslations=false -Ddbuspolicydir=/usr/share/dbus-1/system.d -Ddefault-hierarchy=unified -Ddefault-kill-user-processes=false -Ddefault-locale=C.UTF-8 -Dfallback-hostname=archlinux -Dnologin-path=/usr/bin/nologin '-Dntp-servers=0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org' '-Ddns-servers=1.1.1.1#cloudflare-dns.com 9.9.9.10#dns.quad9.net 8.8.8.8#dns.google 2606:4700:4700::1111#cloudflare-dns.com 2620:fe::10#dns.quad9.net 2001:4860:4860::8888#dns.google' -Drpmmacrosdir=no -Dsysvinit-path= -Dsysvrcnd-path=
The Meson build system
Version: 0.64.0
Source dir: /build/lib32-systemd/src/systemd-stable
Build dir: /build/lib32-systemd/src/build
Build type: native build
Project name: systemd
Project version: 252
C compiler for the host machine: gcc -m32 (gcc 12.2.0 "gcc (GCC) 12.2.0")
C linker for the host machine: gcc -m32 ld.bfd 2.39.0
Host machine cpu family: x86
Host machine cpu: i686
Program tools/add-git-hook.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/add-git-hook.sh)
Message: Activated pre-commit hook
Program tools/meson-build.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/meson-build.sh)
C++ compiler for the host machine: g++ -m32 (gcc 12.2.0 "g++ (GCC) 12.2.0")
C++ linker for the host machine: g++ -m32 ld.bfd 2.39.0
Compiler for C supports arguments -Wno-missing-field-initializers: YES 
Compiler for C supports arguments -Wno-unused-parameter: YES 
Compiler for C supports arguments -Wdate-time: YES 
Compiler for C supports arguments -Wendif-labels: YES 
Compiler for C supports arguments -Werror=format=2: YES 
Compiler for C supports arguments -Werror=format-signedness: YES 
Compiler for C supports arguments -Werror=implicit-function-declaration: YES 
Compiler for C supports arguments -Werror=incompatible-pointer-types: YES 
Compiler for C supports arguments -Werror=int-conversion: YES 
Compiler for C supports arguments -Werror=overflow: YES 
Compiler for C supports arguments -Werror=override-init: YES 
Compiler for C supports arguments -Werror=return-type: YES 
Compiler for C supports arguments -Werror=shift-count-overflow: YES 
Compiler for C supports arguments -Werror=shift-overflow=2: YES 
Compiler for C supports arguments -Werror=undef: YES 
Compiler for C supports arguments -Wfloat-equal: YES 
Compiler for C supports arguments -Wimplicit-fallthrough=5: YES 
Compiler for C supports arguments -Winit-self: YES 
Compiler for C supports arguments -Wlogical-op: YES 
Compiler for C supports arguments -Wmissing-include-dirs: YES 
Compiler for C supports arguments -Wmissing-noreturn: YES 
Compiler for C supports arguments -Wnested-externs: YES 
Compiler for C supports arguments -Wold-style-definition: YES 
Compiler for C supports arguments -Wpointer-arith: YES 
Compiler for C supports arguments -Wredundant-decls: YES 
Compiler for C supports arguments -Wshadow: YES 
Compiler for C supports arguments -Wstrict-aliasing=2: YES 
Compiler for C supports arguments -Wstrict-prototypes: YES 
Compiler for C supports arguments -Wsuggest-attribute=noreturn: YES 
Compiler for C supports arguments -Wunused-function: YES 
Compiler for C supports arguments -Wwrite-strings: YES 
Compiler for C supports arguments -Wno-error=#warnings: NO 
Compiler for C supports arguments -Wno-string-plus-int: NO 
Compiler for C supports arguments -Wno-maybe-uninitialized: YES 
Compiler for C supports arguments -Wno-unused-result: YES 
Compiler for C supports arguments -ftrivial-auto-var-init=zero: YES 
Compiler for C supports arguments -Werror=missing-declarations: YES 
Compiler for C supports arguments -Werror=missing-prototypes: YES 
Compiler for C supports arguments -fdiagnostics-show-option: YES 
Compiler for C supports arguments -fno-common: YES 
Compiler for C supports arguments -fno-strict-aliasing: YES 
Compiler for C supports arguments -fstack-protector: YES 
Compiler for C supports arguments -fstack-protector-strong: YES 
Compiler for C supports arguments -fvisibility=hidden: YES 
Compiler for C supports arguments --param=ssp-buffer-size=4: YES 
Compiler for C supports arguments -ffunction-sections: YES 
Compiler for C supports arguments -fdata-sections: YES 
Compiler for C supports link arguments -Wl,--fatal-warnings: YES 
Compiler for C supports link arguments -Wl,-z,now: YES 
Compiler for C supports link arguments -Wl,-z,relro: YES 
Compiler for C supports link arguments -fstack-protector: YES 
Compiler for C supports link arguments -Wl,--warn-common: YES 
Compiler for C supports link arguments -Wl,--gc-sections: YES 
Compiler for C supports arguments -Wzero-length-bounds: YES 
Checking if "-Werror=shadow with local shadowing" compiles: YES 
Compiler for C++ supports arguments -Wno-missing-field-initializers: YES 
Compiler for C++ supports arguments -Wno-unused-parameter: YES 
Compiler for C supports arguments -Wstringop-truncation: YES 
Checking for size of "dev_t" : 8
Checking for size of "ino_t" : 8
Checking for size of "time_t" : 4
Checking for size of "rlim_t" : 8
Checking for size of "typeof(((struct timex *)0)->freq)" : 4
Checking for size of "char16_t" : 2
Checking for size of "char32_t" : 4
Checking for size of "struct mount_attr" : 32
Checking for size of "struct statx" : 256
Checking for function "secure_getenv" : YES 
Checking for function "__secure_getenv" : NO 
Checking for function "memfd_create" : YES 
Checking for function "gettid" : YES 
Checking for function "pivot_root" : NO 
Checking for function "ioprio_get" : NO 
Checking for function "ioprio_set" : NO 
Checking for function "name_to_handle_at" : YES 
Checking for function "setns" : YES 
Checking for function "renameat2" : YES 
Checking for function "kcmp" : NO 
Checking for function "keyctl" : NO 
Checking for function "copy_file_range" : YES 
Checking for function "bpf" : NO 
Checking for function "statx" : YES 
Checking for function "explicit_bzero" : YES 
Checking for function "reallocarray" : YES 
Checking for function "set_mempolicy" : NO 
Checking for function "get_mempolicy" : NO 
Checking for function "pidfd_send_signal" : NO 
Checking for function "pidfd_open" : NO 
Checking for function "rt_sigqueueinfo" : NO 
Checking for function "mallinfo" : YES 
Checking for function "mallinfo2" : YES 
Checking for function "execveat" : YES 
Checking for function "close_range" : YES 
Checking for function "epoll_pwait2" : YES 
Checking for function "mount_setattr" : YES 
Checking for function "move_mount" : YES 
Checking for function "open_tree" : YES 
Checking for function "fsopen" : YES 
Checking for function "fsconfig" : YES 
Checking for function "getdents64" : YES 
Checking for function "getrandom" : YES 
Configuring version.h using configuration
Program sh found: YES (/usr/bin/sh)
Program echo found: YES (/usr/bin/echo)
Program sed found: YES (/usr/bin/sed)
Program awk found: YES (/usr/bin/awk)
Program stat found: YES (/usr/bin/stat)
Program ln found: YES (/usr/bin/ln)
Program git found: YES (/usr/bin/git)
Program env found: YES (/usr/bin/env)
Program perl found: YES (/usr/bin/perl)
Program rsync found: NO
Program quotaon /usr/sbin/quotaon /sbin/quotaon found: NO
Program quotacheck /usr/sbin/quotacheck /sbin/quotacheck found: NO
Program kmod found: YES (/usr/bin/kmod)
Program kexec /usr/sbin/kexec /sbin/kexec found: NO
Program sulogin found: YES (/usr/bin/sulogin)
Program mount found: YES (/usr/bin/mount)
Program umount found: YES (/usr/bin/umount)
Program loadkeys found: YES (/usr/bin/loadkeys)
Program setfont found: YES (/usr/bin/setfont)
Message: Using /usr/bin/nologin for nologin
Program python3 found: YES (/usr/bin/python3)
Program gperf found: YES (/usr/bin/gperf)
Message: gperf len type is size_t
Has header "sys/capability.h" : YES 
Has header "crypt.h" : YES 
Has header "linux/memfd.h" : YES 
Has header "linux/vm_sockets.h" : YES 
Has header "sys/auxv.h" : YES 
Has header "valgrind/memcheck.h" : NO 
Has header "valgrind/valgrind.h" : NO 
Has header "linux/time_types.h" : YES 
Has header "sys/sdt.h" : YES 
Run-time dependency threads found: YES
Library rt found: YES
Library m found: YES
Library dl found: YES
Library crypt found: YES
Checking if "libatomic" : links: YES 
Checking for function "crypt_ra" with dependency -lcrypt: YES 
Checking for function "crypt_preferred_method" with dependency -lcrypt: YES 
Checking for function "crypt_gensalt_ra" with dependency -lcrypt: YES 
Found pkg-config: /usr/bin/pkg-config (1.8.0)
Run-time dependency libcap found: YES 2.66
Did not find CMake 'cmake'
Found CMake: NO
Run-time dependency libbpf found: NO (tried pkgconfig and cmake)
Run-time dependency mount found: YES 2.38.1
Run-time dependency fdisk found: YES 2.38.1
Run-time dependency pwquality found: NO (tried pkgconfig and cmake)
Run-time dependency libselinux found: NO (tried pkgconfig and cmake)
Run-time dependency libapparmor found: NO (tried pkgconfig and cmake)
Run-time dependency polkit-gobject-1 found: NO (tried pkgconfig and cmake)
Library acl found: YES
Checking for function "crypt_activate_by_token_pin" : NO 
Run-time dependency libcurl found: YES 7.86.0
Run-time dependency libidn2 found: YES 2.3.4
Run-time dependency libqrencode found: NO (tried pkgconfig and cmake)
Library gcrypt found: YES
Library gpg-error found: YES
Run-time dependency gnutls found: YES 3.7.8
Run-time dependency openssl found: YES 3.0.7
Run-time dependency p11-kit-1 found: YES 0.24.1
Run-time dependency libfido2 found: NO (tried pkgconfig and cmake)
Run-time dependency tss2-esys tss2-rc tss2-mu found: YES 3.2.0
3.2.0
3.2.0
Run-time dependency libdw found: YES 0.187
Checking for function "dwelf_elf_e_machine_string" with dependency libdw: YES 
Run-time dependency zlib found: YES 1.2.13
Library bz2 found: YES
Run-time dependency liblzma found: YES 5.2.7
Run-time dependency libzstd found: YES 1.5.2
Run-time dependency xkbcommon found: NO (tried pkgconfig and cmake)
Run-time dependency libpcre2-8 found: YES 10.40
Run-time dependency glib-2.0 found: YES 2.74.1
Run-time dependency gobject-2.0 found: YES 2.74.1
Run-time dependency gio-2.0 found: YES 2.74.1
Run-time dependency dbus-1 found: YES 1.14.4
Program tools/generate-gperfs.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/generate-gperfs.py)
Program tools/make-autosuspend-rules.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/make-autosuspend-rules.py)
Program tools/make-directive-index.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/make-directive-index.py)
Program tools/make-man-index.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/make-man-index.py)
Program tools/meson-render-jinja2.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/meson-render-jinja2.py)
Program tools/update-dbus-docs.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/update-dbus-docs.py)
Program tools/update-man-rules.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/update-man-rules.py)
Program tools/update-hwdb.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/update-hwdb.sh)
Program tools/update-hwdb-autosuspend.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/update-hwdb-autosuspend.sh)
Program tools/update-syscall-tables.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/update-syscall-tables.sh)
Program tools/xml_helper.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/xml_helper.py)
Program tools/dbus_exporter.py found: YES (/build/lib32-systemd/src/systemd-stable/tools/dbus_exporter.py)
Configuring config.h using configuration
Program generate-af-list.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/basic/generate-af-list.sh)
Program generate-arphrd-list.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/basic/generate-arphrd-list.sh)
Program generate-cap-list.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/basic/generate-cap-list.sh)
Program generate-errno-list.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/basic/generate-errno-list.sh)
Program missing_syscalls.py found: YES (/usr/bin/python /build/lib32-systemd/src/systemd-stable/src/basic/missing_syscalls.py)
Program check-filesystems.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/basic/check-filesystems.sh)
Program generate-filesystem-list.py found: YES (/build/lib32-systemd/src/systemd-stable/src/basic/generate-filesystem-list.py)
Program generate-filesystem-switch-case.py found: YES (/build/lib32-systemd/src/systemd-stable/src/basic/generate-filesystem-switch-case.py)
Program sd-journal/generate-audit_type-list.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/libsystemd/sd-journal/generate-audit_type-list.sh)
systemd-stable/src/libsystemd/meson.build:189: WARNING: Project targets '>= 0.53.2' but uses feature introduced in '0.60.0': install_tag arg in custom_target.
Program generate-syscall-list.py found: YES (/build/lib32-systemd/src/systemd-stable/src/shared/generate-syscall-list.py)
Program generate-ip-protocol-list.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/shared/generate-ip-protocol-list.sh)
Program generate-keyboard-keys-list.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/udev/generate-keyboard-keys-list.sh)
Program generate-keyboard-keys-gperf.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/udev/generate-keyboard-keys-gperf.sh)
systemd-stable/src/libudev/meson.build:44: WARNING: Project targets '>= 0.53.2' but uses feature introduced in '0.60.0': install_tag arg in custom_target.
Program test-kernel-install.sh found: YES (/build/lib32-systemd/src/systemd-stable/src/kernel-install/test-kernel-install.sh)
Program generate-dns_type-gperf.py found: YES (/build/lib32-systemd/src/systemd-stable/src/resolve/generate-dns_type-gperf.py)
Compiler for C supports arguments -std=iso9899:2017: YES 
Compiler for C++ supports arguments -std=c++14: YES 
Compiler for C++ supports arguments -std=c++17: YES 
Compiler for C++ supports arguments -std=c++20: YES 
Program generate-sym-test.py found: YES (/build/lib32-systemd/src/systemd-stable/src/test/generate-sym-test.py)
Program test-fstab-generator.sh found: YES (/build/lib32-systemd/src/systemd-stable/test/test-fstab-generator.sh)
Program test-network-generator-conversion.sh found: YES (/build/lib32-systemd/src/systemd-stable/test/test-network-generator-conversion.sh)
Program test-systemctl-enable.sh found: YES (/usr/bin/env bash /build/lib32-systemd/src/systemd-stable/test/test-systemctl-enable.sh)
Program test-systemd-tmpfiles.py found: YES (/build/lib32-systemd/src/systemd-stable/test/test-systemd-tmpfiles.py)
Program hwdb-test.sh found: YES (/build/lib32-systemd/src/systemd-stable/test/hwdb-test.sh)
Configuring test-sysusers.sh using configuration
Program rule-syntax-check.py found: YES (/build/lib32-systemd/src/systemd-stable/test/rule-syntax-check.py)
Program sys-script.py found: YES (/build/lib32-systemd/src/systemd-stable/test/sys-script.py)
Program udev-test.pl found: YES (/build/lib32-systemd/src/systemd-stable/test/udev-test.pl)
Program rpm found: NO
Program rpmspec found: NO
Program test-rpm-macros.sh found: YES (/build/lib32-systemd/src/systemd-stable/test/test-rpm-macros.sh)
Message: Skipping test-rpm-macros since rpm and/or rpmspec are not available
Program udev-dmi-memory-id-test.sh found: YES (/build/lib32-systemd/src/systemd-stable/test/udev-dmi-memory-id-test.sh)
systemd-stable/meson.build:2219: WARNING: Project targets '>= 0.53.2' but uses feature introduced in '0.60.0': install_tag arg in meson.add_install_script.
Compiler for C supports arguments -Wno-maybe-uninitialized: YES (cached)
First supported argument: -Wno-maybe-uninitialized
Message: test-bus-benchmark is a manual test
Message: test-ipv4ll-manual is a manual test
Message: test-acd is a manual test
Message: test-coredump-vacuum is a manual test
Message: test-qcow2 is a manual test
Message: test-inhibit is a manual test
Message: test-session-properties is a manual test
Message: test-patch-uid is a manual test
Message: test-dnssec-complex is a manual test
Message: test-ns is a manual test
Message: Not compiling test-boot-timestamps because ENABLE_EFI is not true
Message: Not compiling test-seccomp because HAVE_SECCOMP is not true
Message: test-ask-password-api is a manual test
Message: test-ipcrm is an unsafe test
Message: test-btrfs is a manual test
Message: Not compiling test-netlink-manual because HAVE_KMOD is not true
Message: Not compiling test-sbat because HAVE_GNU_EFI is not true
Message: test-chase-symlinks is a manual test
Message: test-install is a manual test
Message: test-watchdog is an unsafe test
Message: Not compiling test-utmp because ENABLE_UTMP is not true
Message: test-udev is a manual test
Message: test-nscd-flush is a manual test
Message: Not compiling test-socket-bind because BPF_FRAMEWORK is not true
Checking for size of "long" : 4
Program xsltproc found: YES (/usr/bin/xsltproc)
Message: Skipping binfmt.d.5 because ENABLE_BINFMT is false
Message: Skipping bootctl.1 because HAVE_GNU_EFI is false
Message: Skipping coredump.conf.5 because ENABLE_COREDUMP is false
Message: Skipping coredumpctl.1 because ENABLE_COREDUMP is false
Message: Skipping crypttab.5 because HAVE_LIBCRYPTSETUP is false
Message: Skipping environment.d.5 because ENABLE_ENVIRONMENT_D is false
Message: Skipping homectl.1 because ENABLE_HOMED is false
Message: Skipping homed.conf.5 because ENABLE_HOMED is false
Message: Skipping hostnamectl.1 because ENABLE_HOSTNAMED is false
Message: Skipping hwdb.7 because ENABLE_HWDB is false
Message: Skipping integritytab.5 because HAVE_LIBCRYPTSETUP is false
Message: Skipping journal-remote.conf.5 because HAVE_MICROHTTPD is false
Message: Skipping journal-upload.conf.5 because HAVE_MICROHTTPD is false
Message: Skipping loader.conf.5 because HAVE_GNU_EFI is false
Message: Skipping localectl.1 because ENABLE_LOCALED is false
Message: Skipping loginctl.1 because ENABLE_LOGIND is false
Message: Skipping logind.conf.5 because ENABLE_LOGIND is false
Message: Skipping modules-load.d.5 because HAVE_KMOD is false
Message: Skipping networkctl.1 because ENABLE_NETWORKD is false
Message: Skipping networkd.conf.5 because ENABLE_NETWORKD is false
Message: Skipping oomctl.1 because ENABLE_OOMD is false
Message: Skipping oomd.conf.5 because ENABLE_OOMD is false
Message: Skipping org.freedesktop.home1.5 because ENABLE_HOMED is false
Message: Skipping org.freedesktop.hostname1.5 because ENABLE_HOSTNAMED is false
Message: Skipping org.freedesktop.import1.5 because ENABLE_IMPORTD is false
Message: Skipping org.freedesktop.locale1.5 because ENABLE_LOCALED is false
Message: Skipping org.freedesktop.login1.5 because ENABLE_LOGIND is false
Message: Skipping org.freedesktop.network1.5 because ENABLE_NETWORKD is false
Message: Skipping org.freedesktop.oom1.5 because ENABLE_OOMD is false
Message: Skipping org.freedesktop.portable1.5 because ENABLE_PORTABLED is false
Message: Skipping org.freedesktop.timedate1.5 because ENABLE_TIMEDATED is false
Message: Skipping pam_systemd.8 because HAVE_PAM is false
Message: Skipping pam_systemd_home.8 because ENABLE_PAM_HOME is false
Message: Skipping portablectl.1 because ENABLE_PORTABLED is false
Message: Skipping pstore.conf.5 because ENABLE_PSTORE is false
Message: Skipping repart.d.5 because ENABLE_REPART is false
Message: Skipping runlevel.8 because HAVE_SYSV_COMPAT is false
Message: Skipping sd-login.3 because HAVE_PAM is false
Message: Skipping sd_get_seats.3 because HAVE_PAM is false
Message: Skipping sd_login_monitor_new.3 because HAVE_PAM is false
Message: Skipping sd_pid_get_owner_uid.3 because HAVE_PAM is false
Message: Skipping sd_seat_get_active.3 because HAVE_PAM is false
Message: Skipping sd_session_is_active.3 because HAVE_PAM is false
Message: Skipping sd_uid_get_state.3 because HAVE_PAM is false
Message: Skipping systemd-backlight@.service.8 because ENABLE_BACKLIGHT is false
Message: Skipping systemd-binfmt.service.8 because ENABLE_BINFMT is false
Message: Skipping systemd-bless-boot-generator.8 because HAVE_GNU_EFI is false
Message: Skipping systemd-bless-boot.service.8 because HAVE_GNU_EFI is false
Message: Skipping systemd-boot-system-token.service.8 because HAVE_GNU_EFI is false
Message: Skipping systemd-boot.7 because HAVE_GNU_EFI is false
Message: Skipping systemd-coredump.8 because ENABLE_COREDUMP is false
Message: Skipping systemd-cryptenroll.1 because HAVE_LIBCRYPTSETUP is false
Message: Skipping systemd-cryptsetup-generator.8 because HAVE_LIBCRYPTSETUP is false
Message: Skipping systemd-cryptsetup@.service.8 because HAVE_LIBCRYPTSETUP is false
Message: Skipping systemd-dissect.1 because HAVE_BLKID is false
Message: Skipping systemd-environment-d-generator.8 because ENABLE_ENVIRONMENT_D is false
Message: Skipping systemd-firstboot.1 because ENABLE_FIRSTBOOT is false
Message: Skipping systemd-gpt-auto-generator.8 because HAVE_BLKID is false
Message: Skipping systemd-hibernate-resume-generator.8 because ENABLE_HIBERNATE is false
Message: Skipping systemd-hibernate-resume@.service.8 because ENABLE_HIBERNATE is false
Message: Skipping systemd-homed.service.8 because ENABLE_HOMED is false
Message: Skipping systemd-hostnamed.service.8 because ENABLE_HOSTNAMED is false
Message: Skipping systemd-hwdb.8 because ENABLE_HWDB is false
Message: Skipping systemd-importd.service.8 because ENABLE_IMPORTD is false
Message: Skipping systemd-initctl.service.8 because HAVE_SYSV_COMPAT is false
Message: Skipping systemd-integritysetup-generator.8 because HAVE_LIBCRYPTSETUP is false
Message: Skipping systemd-integritysetup@.service.8 because HAVE_LIBCRYPTSETUP is false
Message: Skipping systemd-journal-gatewayd.service.8 because HAVE_MICROHTTPD is false
Message: Skipping systemd-journal-remote.service.8 because HAVE_MICROHTTPD is false
Message: Skipping systemd-journal-upload.service.8 because HAVE_MICROHTTPD is false
Message: Skipping systemd-localed.service.8 because ENABLE_LOCALED is false
Message: Skipping systemd-logind.service.8 because ENABLE_LOGIND is false
Message: Skipping systemd-measure.1 because HAVE_GNU_EFI is false
Message: Skipping systemd-modules-load.service.8 because HAVE_KMOD is false
Message: Skipping systemd-networkd-wait-online.service.8 because ENABLE_NETWORKD is false
Message: Skipping systemd-networkd.service.8 because ENABLE_NETWORKD is false
Message: Skipping systemd-oomd.service.8 because ENABLE_OOMD is false
Message: Skipping systemd-pcrphase.service.8 because HAVE_GNU_EFI is false
Message: Skipping systemd-portabled.service.8 because ENABLE_PORTABLED is false
Message: Skipping systemd-pstore.service.8 because ENABLE_PSTORE is false
Message: Skipping systemd-quotacheck.service.8 because ENABLE_QUOTACHECK is false
Message: Skipping systemd-random-seed.service.8 because ENABLE_RANDOMSEED is false
Message: Skipping systemd-rc-local-generator.8 because HAVE_SYSV_COMPAT is false
Message: Skipping systemd-repart.8 because ENABLE_REPART is false
Message: Skipping systemd-rfkill.service.8 because ENABLE_RFKILL is false
Message: Skipping systemd-stub.7 because HAVE_GNU_EFI is false
Message: Skipping systemd-sysv-generator.8 because HAVE_SYSV_COMPAT is false
Message: Skipping systemd-time-wait-sync.service.8 because ENABLE_TIMESYNCD is false
Message: Skipping systemd-timedated.service.8 because ENABLE_TIMEDATED is false
Message: Skipping systemd-timesyncd.service.8 because ENABLE_TIMESYNCD is false
Message: Skipping systemd-update-utmp.service.8 because ENABLE_UTMP is false
Message: Skipping systemd-user-sessions.service.8 because HAVE_PAM is false
Message: Skipping systemd-userdbd.service.8 because ENABLE_USERDB is false
Message: Skipping systemd-vconsole-setup.service.8 because ENABLE_VCONSOLE is false
Message: Skipping systemd-veritysetup-generator.8 because HAVE_LIBCRYPTSETUP is false
Message: Skipping systemd-veritysetup@.service.8 because HAVE_LIBCRYPTSETUP is false
Message: Skipping systemd-xdg-autostart-generator.8 because ENABLE_XDG_AUTOSTART is false
Message: Skipping systemd.environment-generator.7 because ENABLE_ENVIRONMENT_D is false
Message: Skipping systemd.netdev.5 because ENABLE_NETWORKD is false
Message: Skipping systemd.network.5 because ENABLE_NETWORKD is false
Message: Skipping sysusers.d.5 because ENABLE_SYSUSERS is false
Message: Skipping telinit.8 because HAVE_SYSV_COMPAT is false
Message: Skipping timedatectl.1 because ENABLE_TIMEDATECTL is false
Message: Skipping timesyncd.conf.5 because ENABLE_TIMESYNCD is false
Message: Skipping userdbctl.1 because ENABLE_USERDB is false
Message: Skipping vconsole.conf.5 because ENABLE_VCONSOLE is false
Message: Skipping veritytab.5 because HAVE_LIBCRYPTSETUP is false
Message: python-lxml not available, not making man page indices
Configuring man using configuration
Configuring html using configuration
Run-time dependency bash-completion found: NO (tried pkgconfig and cmake)
Program jekyll found: NO
Program tools/check-help.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/check-help.sh)
Program tools/check-version.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/check-version.sh)
Program tools/check-directives.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/check-directives.sh)
Compiler for C supports link arguments -fsanitize=address,undefined: YES 
Message: Not compiling fuzz-bootspec:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-bus-match:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-bus-message:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-calendarspec:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-catalog:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-dhcp-client:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-dhcp-server-relay:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-dhcp-server:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-dhcp6-client:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-dns-packet:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-env-file:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-etc-hosts:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-fido-id-desc:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-journal-remote:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-journald-audit:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-journald-kmsg:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-journald-native-fd:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-journald-stream:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-journald-syslog:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-json:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-link-parser:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-lldp-rx:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-ndisc-rs:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-netdev-parser:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-network-parser:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-nspawn-oci:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-nspawn-settings:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-systemctl-parse-argv:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-udev-database:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-udev-rules:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-unit-file:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-varlink:address,undefined because fuzz-tests is set to false
Message: Not compiling fuzz-xdg-desktop:address,undefined because fuzz-tests is set to false
Program tools/git-contrib.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/git-contrib.sh)
Program tools/check-api-docs.sh found: YES (/build/lib32-systemd/src/systemd-stable/tools/check-api-docs.sh)
Build targets in project: 1745
WARNING: Project specifies a minimum meson_version '>= 0.53.2' but uses features which were added in newer versions:
 * 0.60.0: {'install_tag arg in custom_target', 'install_tag arg in meson.add_install_script'}
NOTICE: Future-deprecated features used:
 * 0.55.0: {'ExternalProgram.path'}

systemd 252

    build mode                     : release
    split /usr                     : False
    split bin-sbin                 : False
    prefix directory               : /usr
    rootprefix directory           : /usr
    sysconf directory              : /etc
    include directory              : /usr/include
    lib directory                  : /usr/lib32
    rootlib directory              : /usr/lib32
    SysV init scripts              : 
    SysV rc?.d directories         : 
    PAM modules directory          : /usr/lib32/security
    PAM configuration directory    : /usr/lib/pam.d
    libcryptsetup plugins directory: /usr/lib32/cryptsetup
    RPM macros directory           : no
    modprobe.d directory           : /usr/lib/modprobe.d
    D-Bus policy directory         : /usr/share/dbus-1/system.d
    D-Bus session directory        : /usr/share/dbus-1/services
    D-Bus system directory         : /usr/share/dbus-1/system-services
    D-Bus interfaces directory     : /usr/share/dbus-1/interfaces
    bash completions directory     : /usr/share/bash-completion/completions
    zsh completions directory      : /usr/share/zsh/site-functions
    private shared lib version tag : 252.1-1
    extra start script             : /etc/rc.local
    debug shell                    : /bin/sh @ /dev/tty9
    system UIDs                    : <=999 (alloc >=500)
    system GIDs                    : <=999 (alloc >=500)
    dynamic UIDs                   : 61184…65519
    container UID bases            : 524288…1878982656
    static UID/GID allocations     : tty-gid:5
    /dev/kvm access mode           : 0666
    render group access mode       : 0666
    certificate root directory     : /etc/ssl
    support URL                    : https://lists.freedesktop.org/mailman/listinfo/systemd-devel
    nobody user name               : nobody
    nobody group name              : nobody
    fallback hostname              : archlinux
    default compression method     : zstd
    default DNSSEC mode            : allow-downgrade
    default DNS-over-TLS mode      : no
    default mDNS mode              : yes
    default LLMNR mode             : yes
    default DNS servers            : 1.1.1.1#cloudflare-dns.com
                                     9.9.9.10#dns.quad9.net
                                     8.8.8.8#dns.google
                                     2606:4700:4700::1111#cloudflare-dns.com
                                     2620:fe::10#dns.quad9.net
                                     2001:4860:4860::8888#dns.google
    default NTP servers            : 0.arch.pool.ntp.org
                                     1.arch.pool.ntp.org
                                     2.arch.pool.ntp.org
                                     3.arch.pool.ntp.org
    default cgroup hierarchy       : unified
    default net.naming-scheme value: latest
    default KillUserProcesses value: False
    default locale                 : C.UTF-8
    default nspawn locale          : C.UTF-8
    default status unit format     : description
    default user $PATH             : (same as system services)
    systemd service watchdog       : 3min
    time epoch                     : 1668423099 (2022-11-14T10:51:39+00:00)

  Features
    enabled                        : ACL, SMACK, elfutils, gcrypt, gnutls, libcurl, libfdisk, libidn2, openssl, p11kit, pcre2, tpm2, zstd, xz, zlib, bzip2, initrd, kernel-install, machined, nss-myhostname, nss-mymachines, nss-resolve, nss-systemd, resolve, systemd-analyze, sysupdate, idn, polkit, nscd, dbus, glib, adm group, wheel group, gshadow, link-udev-shared, link-systemctl-shared, link-networkd-shared, link-timesyncd-shared, link-journalctl-shared, link-boot-shared, cryptolib(gcrypt), DNS-over-TLS(openssl)
    disabled                       : AUDIT, AppArmor, IMA, PAM, SECCOMP, SELinux, blkid, libbpf, libcryptsetup, libcryptsetup-plugins, libfido2, libidn, libiptc, microhttpd, pwquality, qrencode, xkbcommon, lz4, backlight, binfmt, bpf-framework, coredump, environment.d, efi, gnu-efi, firstboot, hibernate, homed, hostnamed, hwdb, importd, localed, logind, networkd, oomd, portabled, pstore, quotacheck, randomseed, repart, rfkill, sysext, sysusers, timedated, timesyncd, tmpfiles, userdb, vconsole, xdg-autostart, legacy-pkla, kmod, tpm, man pages, html pages, man page indices, SysV compat, compat-mutable-uid-boundaries, utmp, ldconfig, debug hashmap, debug mmap cache, debug siphash, valgrind, trace logging, install tests, first-boot-full-preset, fexecve, standalone-binaries, coverage, static-libsystemd, static-libudev

  User defined options
    auto_features                  : enabled
    buildtype                      : plain
    libdir                         : /usr/lib32
    libexecdir                     : /usr/lib32
    prefix                         : /usr
    sbindir                        : bin
    wrap_mode                      : nodownload
    b_lto                          : true
    b_pie                          : true
    audit                          : false
    backlight                      : false
    binfmt                         : false
    blkid                          : false
    coredump                       : false
    create-log-dirs                : false
    dbuspolicydir                  : /usr/share/dbus-1/system.d
    default-hierarchy              : unified
    default-kill-user-processes    : false
    default-locale                 : C.UTF-8
    dns-servers                    : 1.1.1.1#cloudflare-dns.com 9.9.9.10#dns.quad9.net 8.8.8.8#dns.google 2606:4700:4700::1111#cloudflare-dns.com 2620:fe::10#dns.quad9.net 2001:4860:4860::8888#dns.google
    efi                            : false
    environment-d                  : false
    fallback-hostname              : archlinux
    firstboot                      : false
    gnu-efi                        : false
    hibernate                      : false
    homed                          : false
    hostnamed                      : false
    html                           : false
    hwdb                           : false
    ima                            : false
    importd                        : false
    kmod                           : false
    ldconfig                       : false
    libcryptsetup                  : false
    libidn2                        : true
    libiptc                        : false
    localed                        : false
    logind                         : false
    lz4                            : false
    machined                       : true
    man                            : false
    microhttpd                     : false
    mode                           : release
    networkd                       : false
    nologin-path                   : /usr/bin/nologin
    nss-myhostname                 : true
    nss-mymachines                 : true
    nss-resolve                    : true
    nss-systemd                    : true
    ntp-servers                    : 0.arch.pool.ntp.org 1.arch.pool.ntp.org 2.arch.pool.ntp.org 3.arch.pool.ntp.org
    oomd                           : false
    pam                            : false
    portabled                      : false
    pstore                         : false
    quotacheck                     : false
    randomseed                     : false
    remote                         : false
    repart                         : false
    resolve                        : true
    rfkill                         : false
    rpmmacrosdir                   : no
    seccomp                        : false
    shared-lib-tag                 : 252.1-1
    sysext                         : false
    sysusers                       : false
    sysvinit-path                  : 
    sysvrcnd-path                  : 
    timedated                      : false
    timesyncd                      : false
    tmpfiles                       : false
    tpm                            : false
    translations                   : false
    userdb                         : false
    utmp                           : false
    vconsole                       : false
    version-tag                    : 252.1-1-arch
    xdg-autostart                  : false
```

For any reason `HAVE_TPM2` is set. Will check how that is evaluated...

eworm-de commented:
Oh, there is another meson option `tpm2`... Did not know about that. I guess things work as expected if I set this to `false`.

Still I think this should be auto-detected.

DaanDeMeyer commented:
> Still I think this should be auto-detected.

That's what I'm confused about, meson seems to think that the tpm2 libraries are enabled, since the tpm2 option is listed under the enabled features, yet you say the test is failing, which indicates the library is missing.

eworm-de commented:
Ah, wait! The build environment has the package `tpm2-tss` installed (as `libsecret` depends on it). Looks like meson finds that, but `dlopen` fails as x86_64 code can not be loaded in a multilib (32 bit / i686) build. How and where does it check for the library?

DaanDeMeyer commented:
The check happens here: https://github.com/systemd/systemd/blob/main/meson.build#L1487

eworm-de commented:
And that uses `pkgconf` and friends? It is right in finding the package then, it just does not notice the wrong dependency.

For now I dropped my workaround with `sed` and just added `-Dtpm2=false` to my meson options. I am fine with it that way... Should we just close the issue?

DaanDeMeyer commented:
Let's close for now I guess, this seems like more of a meson issue
